### ISSUE TITLE
Previous code Commit's are copying to new branch 

### ISSUE THREAD

**Issue by vickysmart15** [microsoft/vscode#199136]
Type: <b>Bug</b>

Hi, 

 I have been recently facing an issue, that the commits of previous branch are being copied to new branch. 

When i commit from new branch, I could see the previous commits of another branch to the new branch in my pull request. 

This affecting my productivity. 

Please fix this.  Is there anything in VSC to clear the cache ??

Thanks. 

Vignesh

VS Code version: Code 1.84.2 (1a5daa3a0231a0fbba4f14db7ec463cf99d7768e, 2023-11-09T10:51:52.184Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Remote OS version: Linux x64 5.15.133.1-microsoft-standard-WSL2

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|12th Gen Intel(R) Core(TM) i7-1265U (12 x 2688)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.69GB (18.43GB free)|
|Process Argv|--folder-uri=vscode-remote://wsl+Ubuntu-22.04/home/vignesh15/latest_repo/sky-newCRM-infrastructure --remote=wsl+Ubuntu-22.04 --crash-reporter-id a8ac8c6e-52d5-4f84-88b6-4bb034aae137|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu-22.04|
|OS|Linux x64 5.15.133.1-microsoft-standard-WSL2|
|CPUs|12th Gen Intel(R) Core(TM) i7-1265U (12 x 2687)|
|Memory (System)|15.47GB (14.25GB free)|
|VM|0%|
</details><details><summary>Extensions (14)</summary>

Extension|Author (truncated)|Version
---|---|---
remotehub|Git|0.62.0
remote-ssh|ms-|0.107.0
remote-ssh-edit|ms-|0.86.0
remote-wsl|ms-|0.81.8
azure-repos|ms-|0.38.0
remote-explorer|ms-|0.4.1
remote-repositories|ms-|0.40.0
jinja|who|0.0.8
gitlens|eam|14.5.1
EditorConfig|Edi|0.16.4
terraform|has|2.29.0
vscode-docker|ms-|1.28.0
vscode-yaml|red|1.14.0
json|Zai|2.0.2

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
py29gd2263cf:30880073
vscaat:30438848
vsclangdc:30486549
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyonecf:30548226
2e4cg342:30602488
f6dab269:30613381
showlangstatbar:30737416
pythonfmttext:30731395
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
pythonnosmt12:30797651
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
dsvsc013:30795093
dsvsc014:30804076
dsvsc015:30845448
pythontestfixt:30871694
pythonregdiag2cf:30871583
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
accentitlementsc:30887149
dsvsc016:30886110
dsvsc017cf:30886113
dsvsc018cf:30886115
aa_t_chat:30882232

```

</details>

**Comment by joaomoreno** [comment]
@vickysmart15 Can you share a [screen recording](https://gifcap.dev/) of your issue?

**Comment by VSCodeTriageBot** [comment]
This issue has been closed automatically because it needs more information and has not had recent activity. See also our [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.

Happy Coding!

**Comment by bjarketrux** [comment]
I have tried something similar twice.

Steps to reproduce:
* Have a task branch with commits (pushed)
* Have a local `develop` behind the server
* When on the task branch change branch to develop (I use the status bar branch selector)
* See that your local `develop` it behind (but you have no local commits) and click the sync button
* **Observe** Your local `develop` is updated to latest version of `develop`, BUT commits from the previous branch is synched to your local develop

I needed to delete my local `develop` branch and pull it again to continue (not a big deal, but you get a bit confused).
Will try to make a screen recording of it next time I switch to develop.
